FWCore/Framework/src/SubProcess.cc

```diff
@@ -498,9 +508,54 @@ namespace edm {
     propagateProducts(InProcess, parentPrincipal, processBlockPrincipal);
 
     ProcessBlockTransitionInfo transitionInfo(processBlockPrincipal);
-    using Traits = OccurrenceTraits<ProcessBlockPrincipal, BranchActionGlobalEnd>;
-    endGlobalTransitionAsync<Traits>(
-        std::move(iHolder), *schedule_, transitionInfo, serviceToken_, subProcesses_, cleaningUpAfterException);
+
+    if (parentProducedProductIsKept(parentPrincipal, processBlockPrincipal)) {
+      auto runEndProcessBlock =
+          make_waiting_task([this, iWait = std::move(iHolder), info = transitionInfo, cleaningUpAfterException](
```

Comment:
I think the `iHolder` can not be moved here, because it is used below in this scope. The uses seem to be just for the group, so something along the following could work
```suggestion
      auto& taskGroup = *iHolder.group();
      auto runEndProcessBlock =
          make_waiting_task([this, iWait = std::move(iHolder), info = transitionInfo, cleaningUpAfterException](
```
and later
```diff
- WaitingTaskHolder holder(*iHolder.group(), runEndProcessBlock);
+ WaitingTaskHolder holder(taskGroup, runEndProcessBlock);
```
while keeping the `std::move()` here. (although holding a pointer/reference returned by moved-from object looks suspicious, on the other hand I think we have that kind of code already somewhere)

Reply:
Good catch. I used your proposed fix and it seems to work well.

I agree this needed to be fixed and was very prone to causing an error in the future. Although, it technically worked and all the tests passed because the WaitingTaskHolder move constructor does not modify the tbb::task_group pointer of the moved-from object. As long as that remained true, it would have continued to work.